contracts/core/MigrationRibbon.sol

```diff
@@ -0,0 +1,169 @@
+
+    uint16 public lastRound;
+
+    mapping(uint16 => uint256) internal amounts;
+
+    // ----------------------------------------------------------
+    // -----------------------  EVENTS  -------------------------
+    // ----------------------------------------------------------
+
+    event Deposit(address indexed recipient, uint256 amount);
+
+    event Withdraw(uint16 indexed round);
+
+    // ----------------------------------------------------------
+    // -----------------------  ERRORS  -------------------------
+    // ----------------------------------------------------------
+
+    error NotEnoughFunds();
+
+    error RoundNotCompleted();
+
+    error FundsLastRoundNotWithdrawn();
+
+    // ----------------------------------------------------------
+    // -----------------------  GETTERS  ------------------------
+    // ----------------------------------------------------------
+
+    function getCurrentRound() public view returns (uint16) {
+        return ribbonVault.vaultState().round;
+    }
+
+    // ----------------------------------------------------------
+    // -----------------------  SETTERS  ------------------------
+    // ----------------------------------------------------------
+
+    function setNeuronCollateralVault(address _neuronCollateralVault) external {
+        neuronCollateralVault = NeuronCollateralVaultInterface(_neuronCollateralVault);
+    }
+
+    // ----------------------------------------------------------
+    // ---------------------  INITIALIZE  -----------------------
+    // ----------------------------------------------------------
+
+    function initialize(address _ribbonVault, address _neuronCollateralVault) public initializer {
+        // ??? need zero-address validation?
+
+        // Inherited
+        __UUPSUpgradeable_init();
+        __Ownable_init();
+
+        // Ribbon
+        ribbonVault = RibbonVaultInterface(_ribbonVault);
+        ribbonAssetToken = IERC20(ribbonVault.vaultParams().asset);
+        lastRound = getCurrentRound();
+
+        // Neuron
+        neuronCollateralVault = NeuronCollateralVaultInterface(_neuronCollateralVault);
+    }
+
+    // ----------------------------------------------------------
+    // -----------------------  PROXY  --------------------------
+    // ----------------------------------------------------------
+
+    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {
+        // Proxy upgrade callback that uses validating modifier [onlyOwner]. See implementation [UUPSUpgradeable]
+    }
+
+    // ----------------------------------------------------------
+    // ----------------------  ACTIONS  -------------------------
+    // ----------------------------------------------------------
+
+    // Before call approve [_amount] ribbonVault(ERC20) token
+    function deposit(uint256 _amount) external {
+        uint16 currentRound = getCurrentRound();
+
```

Comment:
Check if `_neuronCollateralVault` is true in mapping, otherwise throw Error "NotAuthorizedNeuronCollateralVault`